protocol/contracts/governance/ElderElection.sol

```diff
@@ -0,0 +1,149 @@
+//SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import "@openzeppelin/contracts/utils/cryptography/draft-EIP712.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "./Templar.sol";
+
+/**
+ * @title A contract recording voting in the election of temple elders.
+ *
+ * @notice
+ * The election runs "continuously" - timing for when the results are actually
+ * actioned is handled at the social/policy layer.
+ * 
+ * Any address can vote by endorsing candidates. However only endorsements from voters
+ * who hold temple (aka "token stakeholders") or who are temple team members (aka "impact
+ * stakeholders") have vote value.
+ *
+ * EIP712 structured data signing supports 'gasless voting' via a relay
+ */
+contract ElderElection is EIP712, Ownable, AccessControl {
```

Comment:
I wonder if we should add Pausable contract to this, such that (if need be) we can pause the election voting when we want to snapshot at the policy layer?

Reply:
Given that reading votes etc requires processing the emitted events, I thought we would just read the events up to a given time to get the result at that time. 

Happy to add pausing if needed.